File: app/src/main/java/com/wpapper/iping/ui/utils/L.java
package com.wpapper.iping.ui.utils;

import android.util.Log;

import iping.wpapper.com.iping.BuildConfig;


public class L {

    private static final String TAG_FORMAT = "%s-%s(%d)";

    static boolean DEBUG = BuildConfig.DEBUG;

    public static void i(String message) {
        if (DEBUG) {
            Log.i(tag(), message == null ? " NO MESSAGE " : message);
        }
    }

    public static void e(Exception e) {
        if (DEBUG) {
            Log.e(tag(), e.getMessage() == null ? " NO MESSAGE " : e.getMessage(), e);
        }
    }

    public static void e(String message) {
        if (DEBUG) {
            Log.e(tag(), message == null ? " NO MESSAGE " : message);
        }
    }

    private static String tag() {
        Throwable throwable = new Throwable();
        StackTraceElement target = throwable.getStackTrace()[2];
        if (target == null) {
            return "UNKOWN";
        }
        return String.format(TAG_FORMAT, target.getClassName().substring(target.getClassName().lastIndexOf('.') + 1), target.getMethodName(), target.getLineNumber());
    }

}

File: app/src/main/java/com/wpapper/iping/ui/utils/L.kt
package com.wpapper.iping.ui.utils

import android.util.Log
import com.wpapper.iping.BuildConfig


object L {

    private val TAG_FORMAT = "%s-%s(%d)"

    internal var DEBUG = BuildConfig.DEBUG

    fun i(message: String?) {
        if (DEBUG) {
            Log.i(tag(), message ?: " NO MESSAGE ")
        }
    }

    fun e(e: Exception) {
        if (DEBUG) {
            Log.e(tag(), if (e.message == null) " NO MESSAGE " else e.message, e)
        }
    }

    fun e(message: String?) {
        if (DEBUG) {
            Log.e(tag(), message ?: " NO MESSAGE ")
        }
    }

    private fun tag(): String {
        val throwable = Throwable()
        val target = throwable.stackTrace[2] ?: return "UNKOWN"
        return String.format(TAG_FORMAT, target.className.substring(target.className.lastIndexOf('.') + 1), target.methodName, target.lineNumber)
    }

}
